### PR TITLE
MAINT: Remove PY3

### PR DESCRIPTION
**Pull request by bashtage**
Remove PY3 and related shims

- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

**Comment by bashtage** [comment]
ping @jbrockmendel

File: statsmodels/datasets/utils.py [hunk 1]
```diff
@@ -114,28 +114,16 @@ def _get_cache(cache):
...
-        import zlib
-        # use protocol 2 so can open with python 2.x if cached in 3.x
-        data = data.decode('utf-8')
-        open(cache_path, "wb").write(zlib.compress(cPickle.dumps(data,
-                                                                 protocol=2)))
-    else:
-        open(cache_path, "wb").write(cPickle.dumps(data).encode("zip"))
+    import zlib
+    data = data.decode('utf-8')
+    open(cache_path, "wb").write(zlib.compress(cPickle.dumps(data)))
```
Comment (jbrockmendel) [inline statsmodels/datasets/utils.py]: should this be a `with open`?  don't want to rely on gc to close the file.  fine to leave for follow up

File: statsmodels/genmod/tests/test_glm_weights.py [hunk 2]
```diff
@@ -25,9 +25,8 @@
 TestBinomialVsVarWeights          statsmodels.GLM        X      X                     X                                                                                                  bfgs
 TestGlmGaussianWLS                statsmodels.WLS        X      X                     X                                                                                                  bfgs
 ================================= ====================== ====== ===================== === ======= ======== ============== ============= ============== ============= ============== ==== =========
-"""
+"""  # noqa: E501
 from __future__ import division
```
Comment (jbrockmendel) [inline statsmodels/genmod/tests/test_glm_weights.py]: these are no longer needed right?

File: statsmodels/genmod/tests/test_glm_weights.py [hunk 3]
```diff
@@ -779,7 +778,7 @@ def test_warnings_raised():
     cov_kwds = {'groups': gid, 'use_correction': False}
 
     # Work around for buggy pytest repeated warning capture on Python 2.7
-    warning_type = SpecificationWarning if PY3 else None
+    warning_type = SpecificationWarning
```
Comment (jbrockmendel) [inline statsmodels/genmod/tests/test_glm_weights.py]: comment above unnecessary

File: statsmodels/genmod/tests/test_glm_weights.py (hunk 2 above)
Comment (bashtage) [reply statsmodels/genmod/tests/test_glm_weights.py]: True.

File: statsmodels/iolib/foreign.py [hunk 4]
```diff
@@ -1023,23 +1012,24 @@ def genfromdta(fname, missing_flt=-999., encoding=None, pandas=False,
                 i = col
                 col = data.columns[col]
                 data[col] = data[col].apply(_stata_elapsed_date_to_datetime,
-                        args=(fmtlist[i],))
+                                            args=(fmtlist[i],))
     elif convert_dates:
-        #date_cols = np.where(map(lambda x : x in _date_formats,
+        # date_cols = np.where(map(lambda x : x in _date_formats,
```
Comment (jbrockmendel) [inline statsmodels/iolib/foreign.py]: this should be deleted

File: statsmodels/datasets/utils.py (hunk 1 above)
Comment (bashtage) [reply statsmodels/datasets/utils.py]: Probably, was doing a mechanical for this pass

File: statsmodels/genmod/tests/test_glm_weights.py (hunk 3 above)
Comment (bashtage) [reply statsmodels/genmod/tests/test_glm_weights.py]: Should look for "Python 2.7" or just " 2.7" or "2.7 "

**Comment by jbrockmendel** [comment]
small comments, generally looks really nice.  some more things can be pruned from compat.python in follow-ups

**Pull request merged**